rm/rm-node-updater/src/main/java/org/ow2/proactive/resourcemanager/updater/RMNodeUpdater.java

```diff
@@ -46,113 +74,496 @@
+            }
         }
-        System.out.println("Launching a computing node");
-        Class<?> cls = Class.forName("OneJar");
-        Method meth = cls.getMethod("main", String[].class);
-        meth.invoke(null, (Object) args);
+        return standardPropertySet;
     }
+
+    /**
+     * Returns a set containing all internal or sun-proprietary java system properties, which will not be forwarded to the new JVM
+     * @return set of property names
+     */
+    private Set<String> allInternalProperties() {
+        Set<String> internalPropertySet = new HashSet<>();
+        for (String propertyName : System.getProperties().stringPropertyNames()) {
+            if (propertyName.startsWith("sun.") || (propertyName.startsWith(XTRA_OPTION))) {
+                internalPropertySet.add(propertyName);
+            } else {
+                switch (propertyName) {
+                    case "file.encoding.pkg":
+                    case "user.script":
+                    case "user.country":
+                    case "java.runtime.version":
+                    case "java.awt.graphicsenv":
+                    case "java.endorsed.dirs":
+                    case "user.variant":
+                    case "user.timezone":
+                    case "java.runtime.name":
+                    case "java.vendor.url.bug":
+                    case "java.security.manager":
+                    case "java.awt.printerjob":
+                    case "awt.toolkit":
+                    case "java.vm.info":
+                        internalPropertySet.add(propertyName);
+                        break;
+                    default:
+                        // do nothing
+                }
+
+            }
+        }
+        return internalPropertySet;
+    }
+
+    /**
+     * Builds a list of options which converts java properties starting with XTRA_OPTION prefix to java command line arguments
+     * @return a list of command line options
+     * @link RMNodeUpdater.XTRA_OPTION
+     */
+    private List<String> allNonStandardXOptionsConvertedToProperties() {
+        List<String> xOptions = new ArrayList<>();
+        for (String propertyName : System.getProperties().stringPropertyNames()) {
+            if (propertyName.startsWith(XTRA_OPTION)) {
+                xOptions.add("-" + System.getProperty(propertyName));
+            }
+        }
+        return xOptions;
+    }
+
+    private static void trustEveryone() {
+        try {
+            HttpsURLConnection.setDefaultHostnameVerifier(new HostnameVerifier() {
+                public boolean verify(String hostname, SSLSession session) {
+                    return true;
+                }
+            });
+            SSLContext context = SSLContext.getInstance("TLS");
+            context.init(null, new X509TrustManager[] { new X509TrustManager() {
+                public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)